### ISSUE TITLE
sqlitedict freezes when there is Error in one of the threads

### ISSUE THREAD

**Issue by ziky90** [piskvorky/sqlitedict#24]
When I am trying to access sqlite via sqlitedict with some unsupported type whole system freezes and it's even impossible to break it with ctrl + c. 

It is happening with following stack trace:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 813, in __bootstrap_inner
    self.run()
  File "/home/hadoop/.vew/iab/lib/python2.7/site-packages/sqlitedict.py", line 270, in run
    cursor.execute(req, arg)
InterfaceError: Error binding parameter 0 - probably unsupported type.  
```

It seems that there is thrown an Error in one thread, but other threads keeps working and probably hangs  in the system.

**Comment by jquast** [comment]
I've been investigating this today while writing test cases for #25 in particular (also your issue).

(Of course, if we just resolve the exception itself, we can negate the need to handle this specific condition).

sqlitedict uses an asynchronous "fire and forget" model:

```python
    def execute(self, req, arg=None, res=None):
        """
        `execute` calls are non-blocking: just queue up the request and return immediately.
        """
        self.reqs.put((req, arg or tuple(), res))
```

Although the exception could be caught and communicated by an event/semaphore/shared variable to the MainThread, there is no place in the MainThread to identify and report it!

Just throwing this out there if anyone has any thoughts or experience in this matter.

**Comment by jquast** [comment]
@ziky90 if you care to test https://github.com/jquast/sqlitedict/tree/issue-24-freezes-on-InterfaceError and comment on https://github.com/piskvorky/sqlitedict/pull/28 the inner exception and stacktrace, and original outer stacktrace should be printed to ERROR log, and an exception raised in the parent thread (eventually).

**Comment by piskvorky** [comment]
Fixed by #28 .
